src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs

```diff
@@ -125,6 +125,21 @@ public StackTrace(StackFrame frame)
             _numOfFrames = 1;
         }
 
+        /// <summary>
+        /// Constructs a "fake" stack trace
+        /// </summary>
+        public StackTrace(IEnumerable<StackFrame> frames)
+        {
+            ArgumentNullException.ThrowIfNull(frames);
+
+            List<StackFrame> frameList = new List<StackFrame>();
+            foreach (var frame in frames)
+                frameList.Add(frame);
```

Comment:
>frame [](http://example.com/codeflow?start=30&length=5)

Do we need validate or skip the null frames? I am seeing we don't do that in the constructor `public StackTrace(StackFrame frame)` but raising the question if we need to do better here.

Reply:
@tarekgh - What would the BCL usually do when taking an array of reference typed objects as an input parameter? If we would normally null check them then it seems reasonable to null check them here.

Reply:
I think we don't have a specific guideline for validating the items of the `IEnumerable<T>` parameter. @bartonjs may have further input on this matter. Upon examining APIs such as `String.Join()` and `StringBuilder.AppendJoin()`, it appears that they simply disregard null items when processing the provided list.